shotover-proxy/src/frame/cassandra.rs

```diff
@@ -34,7 +34,7 @@ fn get_batch_len(bytes: &[u8]) -> Result<NonZeroU32> {
     }
 
     let short_bytes = &bytes[1..3];
-    let short = u16::from_be_bytes(short_bytes.try_into()?);
+    let short = u16::from_be_bytes(short_bytes.try_into()?).max(1); // it is valid for a batch statement to have 0 statements, but for the purposes of shotover throttling we can count it as one query
```

Comment:
I feel like I'm missing something, but I need to ask... should this be min? How do we ever get a batch len longer than 1 in this case?


I thought about commenting on this on the other PR. But here is my thoughts as this will have far ranging consquences to all transform authors for only the silliest of non-real world queries (batches used properly should never be 0 or 1 despite being valid in the protocol). 

We may need to distinguish how batches work in relation to query counts and define formally what a batch is to shotover. To this I would simply say we formally define that shotover will always count empty batches as a batch size of one. So transform authors need to be aware of this, if they care.

They can also get the real count from the AST if needed

Reply:
Hmmm I havent checked if min or max is the correct one here but we should probably just replace it with `NonZeroU32::new(short).unwrap_or(nonzero!(1u32))`
The advantages are:
* its more obviously correct
* unwraps can have slightly worse performance due to the compiler failing to optimize out the panic. Not something thats usually worth worrying about but it adds another bullet point here :P

Reply:
> batches used properly should never be 0 or 1 despite being valid in the protocol

I think its not too unreasonable for a programmatically generated batch to contain 0 or 1 messages.

> We may need to distinguish how batches work in relation to query counts and define formally what a batch is to shotover. To this I would simply say we formally define that shotover will always count empty batches as a batch size of one. So transform authors need to be aware of this, if they care.

I guess the outcome of this is `Message::message_count` needs a doc comment that explains the meaning of the returned value and how empty batches are defined,  and in a way that doesnt refer to throttling as there are other possible use cases.

Reply:
So we want to make `Message::get_message_count` available to transform authors? In that case it should be named a bit better, we might want to call it `Message::get_request_count` instead and define what a request is in Shotover instead. (i.e. every request except for batches counts as 1 request.